src/v/cloud_storage/partition_manifest.cc

```diff
@@ -2710,4 +2713,11 @@ void partition_manifest::process_anomalies(
       _last_scrubbed_offset);
 }
 
+std::ostream& operator<<(std::ostream& o, const partition_manifest& pm) {
+    o << "{manifest: ";
+    pm.serialize_json(o, false);
+    o << "; last segment: " << pm.last_segment() << "}";
```

Comment:
The `last_segment` returns optional. Is the case when it returns `nullopt` handled correctly? (It probably is handled in the overload of the `operator << for optoinal<T>` but it's better to check).

Reply:
seems to work:
```
DEBUG 2024-12-10 14:17:01,961 [shard 0:main] cloud_storage - [fiber0~0|0|299993ms [kafka/topic-19/0, rev: 79]] - partition_recovery_manager.cc:322 - Partition manifest used for recovery: {manifest: {"version":1,"namespace":"kafka","topic":"topic-19","partition":0,"revision":74,"last_offset":374902,"insync_offset":374911,"start_offset":0,"cloud_log_size_bytes":223367602,"last_partition_scrub":1733840181595,"highest_producer_id":20}; last segment: {{o=372985-374902 t={timestamp: 1733840199312}-{timestamp: 1733840199359}}}}
```

Reply:
but if the manifest is empty the `pm.last_segment()` will return nullopt, will it work in this case?

Reply:
Yes it will:
`DEBUG 2024-12-14 00:03:31,026 [shard 0:main] partition_manifest_test - partition_manifest_test.cc:2688 - {manifest: {"version":1,"namespace":"","topic":"","partition":-2147483648,"revision":-9223372036854775808,"last_offset":0,"cloud_log_size_bytes":0}; last segment: {nullopt}}`